api/graphql/schema.graphql

```diff
@@ -181,7 +181,7 @@ type Applicant {
     # Gov info
     isGovEmployee: Boolean @rename(attribute: "is_gov_employee")
     govEmployeeType: GovEmployeeType @rename(attribute: "gov_employee_type")
-    interestedInLaterOrSecondment: Boolean @rename(attribute: "interested_in_later_or_secondment")
+    interestedInLaterOrSecondment: Boolean  @deprecated(reason: "Replaced by role and salary expectations") @rename(attribute: "interested_in_later_or_secondment")
```

Comment:
Not sure I understand the reason to deprecate instead of remove. Is there a reason not to remove the column `interested_in_later_or_secondment` in the database as well as in the model if we aren't running any migrations to move the existing data into other columns?

Reply:
I wasn't sure how to migrate the data appropriately since it is going from a `boolean` to an `ENUM`.

If we are certain we won't need it any longer I can remove it entirely and write a migration to drop the column. I do guess it's not a huge deal since this field should be behind a feature flag everywhere it appears. 🤔

Reply:
This has been completely removed with [completely remove secondment field](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3698/commits/1c5626552f3a90dfc8a1de0fef1291e05f889127)